pkg/controller/snapshot_controller.go

```diff
@@ -1068,3 +1089,117 @@ func (ctrl *csiSnapshotController) removeSnapshotFinalizer(snapshot *crdv1.Volum
 	klog.V(5).Infof("Removed protection finalizer from volume snapshot %s", snapshotKey(snapshot))
 	return nil
 }
+
+// addSnapshotSourceFinalizer adds a Finalizer for VolumeSnapshot Source PVC.
+func (ctrl *csiSnapshotController) addSnapshotSourceFinalizer(snapshot *crdv1.VolumeSnapshot) error {
```

Comment:
I think you can merge `addSnapshotSourceFinalizer` and `needToAddSnapshotSourceFinalizer` together into e.g. `ensureSnapshotSourceFinalizer` and save one `getClaimFromVolumeSnapshot` and couple of common lines of code.

Reply:
Sure.  Merged the two functions into one.